### PR TITLE
[WIP] 'Simple' Frame tests and fixes

### PR DESCRIPTION
**Pull request by Funth0mas**
* Regression Frame tests for the obvious functionality
* try to resolve problems found:
  * fix J1939 getter/setter (probably leftover after introducing attrs?)
  * fix attribute() 'default' value (introduced by me :-( )
* Improve two docstrings.

Thanks for review!

### CODE REVIEW

File: src/canmatrix/canmatrix.py [hunk 1]
```diff
@@ -710,9 +708,16 @@ def calcDLC(self):
 
     def findNotUsedBits(self):
         """
-        Find unused bits in frame
+        Find unused bits in frame.
+
+        Represents the bit usage in the frame by means of a list with 64 items.
+        Every item represents one bit and contains index of signal, occupying that bit.
```
Comment (altendky) [inline src/canmatrix/canmatrix.py]: Sadly, it simply doesn't contain 'the index' of the signal, it contains the index plus 1...  But, this is a code bug from my perspective.  #227  I suppose that in the mean time the docs should represent the actual functionality.

File: src/canmatrix/canmatrix.py [hunk 2]
```diff
@@ -676,7 +674,7 @@ def addAttribute(self, attribute, value):
         """
         try:
             self.attributes[attribute] = str(value)
-        except UnicodeDecodeError:
+        except UnicodeDecodeError:  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (Funth0mas) [reply src/canmatrix/canmatrix.py]: I removed the pragma.
The `str` can raise this exception in py2. Eduard stated somewhere in #197, that this exception must have be caught for 'some' reason - maybe wrong coding in an input file. But I think this is the only occurence of such `str()` in the whole file, strange.

**Comment by Funth0mas** [comment]
I committed only the merge + resolve conflicts, not the code I wanted to commit. I'll do it later, sorry.

File: src/canmatrix/canmatrix.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (altendky) [reply src/canmatrix/canmatrix.py]: Someday we'll worry about full coverage and have to deal with this.  :]  Maybe by then we can have a separate thing to correct the crazy encoding scenarios.

**Comment by altendky** [comment]
@ebroecker, this has my approval albeit somewhat weakly since I'm not particularly familiar with the 1939 stuff.  Because of that I'm leaving it to you to make the final merge decision.

**Pull request merged**